### PR TITLE
Ignore ecma2yml config item if source xml folder does not exist

### PR DESCRIPTION
**Pull request by live1206**
[#AB304055](https://dev.azure.com/ceapex/Engineering/_workitems/edit/304055)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6574)

### CODE REVIEW

**Review by TianqiZhang — APPROVED** [review]

File: src/docfx/config/ops/OpsConfigLoader.cs [hunk 1]
```diff
@@ -132,7 +132,12 @@ private static (JObject obj, string path, string name)[] GetDependencies(OpsConf
                 foreach (var ecma2Yaml in ecma2YamlConfig)
                 {
                     var ecma2YamlJObject = JsonUtility.ToJObject(ecma2Yaml);
-                    ecma2YamlJObject[nameof(ECMA2YamlRepoConfig.SourceXmlFolder)] = Path.GetRelativePath(buildSourceFolder, ecma2Yaml.SourceXmlFolder);
+                    var sourceXmlFolder = Path.GetRelativePath(buildSourceFolder, ecma2Yaml.SourceXmlFolder);
+                    if (!Directory.Exists(Path.GetFullPath(Path.Combine(docsetPath, sourceXmlFolder))))
```
Comment (TianqiZhang) [inline src/docfx/config/ops/OpsConfigLoader.cs:136]: sourceXmlFolder is not always relative to docsetPath, it could be relative to repo root.

Comment (live1206) [reply src/docfx/config/ops/OpsConfigLoader.cs:136]: As offline discussed, we get the relative path to `buildSourceFolder` firstly, which is `docsetPath`.

Comment (yufeih) [inline src/docfx/config/ops/OpsConfigLoader.cs:136]: Consider check existence before calling EcmaToYaml? I would like the config adapter to only to config JSON conversion so if a user writes it in `docfx.yml` this directory check is still there?

Comment (live1206) [reply src/docfx/config/ops/OpsConfigLoader.cs:136]: Then we need to check the existence in multiple places, such as `SourceMap`

Comment (yufeih) [reply src/docfx/config/ops/OpsConfigLoader.cs:136]: yep, `FileResolver.TryReadString` seems like a reasonable addition to have.

**Review by yufeih — APPROVED** [review]
👍

**Pull request merged**